distro/etc_nixos/overlays/test-asterinas/test-framework.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/sh
```

Comment:
This script is definitely a great help to write test programs. But this script alone is not a test framework. What we want is to set up the convention/framework to add more tests for Nix packages. Think of Python, Go, QEMU, and more. We will need to have a proper README.md to describe how to add new tests. This is missing from this PR.

If this documentation was written, then one would find that at least three locations must be modified separately to add a new test:
* `distro/etc_nixos/overlays/test-asterinas/`
* `distro/etc_nixos/configs/`
* `.github/workflows/test_nixos_full.yml`

Don't you think that the information about a Nix package test is spread over too many locations?